### PR TITLE
controllers: add `DisableSpec` and `DisableStatus` behavior to `KubernetesApply`

### PR DESCRIPTION
**Pull request by lizzthabet**
Add logic to watch the `configmap` specified in a `KubernetesApply`'s `DisableSpec` and update its `DisableStatus`, plus delete any of its managed objects when it's disabled.

Note: there's a `ForceApply` that runs during a `Reconcile` when a `KubernetesApply` isn't disabled. It will overwrite any `DisableStatus` that's been updated earlier in the reconciling, so I added a couple lines to make sure that status gets written properly. If there's a better way to handle this, lmk!

Closes [CH12731](https://app.shortcut.com/windmill/story/12731/rest-of-api-controllers-other-than-cmd-s-deal-with-disablesource-status)

### CODE REVIEW

**Comment by nicks** [comment]
what's the current status PR (mainly just want to make sure you're not blocked on a review)

File: internal/controllers/core/kubernetesapply/reconciler.go
Comment (lizzthabet) [reply internal/controllers/core/kubernetesapply/reconciler.go:130]: ok! that makes sense and is really helpful context. i'll update this and the `FileWatch` dispatches to happen before the disabling happens. thank you both for prodding on the placement of these (soon-to-no-longer-be) redundant dispatches.

File: internal/controllers/core/filewatch/controller.go
Comment (lizzthabet) [reply internal/controllers/core/filewatch/controller.go]: this is fixed now!

**Comment by lizzthabet** [comment]
thanks to @landism for helping me debug the failing test! i documented the preexisting issue that caused the failing test in #5125, but i believe that issue shouldn't affect (and is out of scope) of this pr now. @nicks and @landism, lmk if you have any other feedback before this is ready to merge.

**Review by nicks — APPROVED** [review]

File: internal/controllers/core/kubernetesapply/reconciler.go
Comment (landism) [inline internal/controllers/core/kubernetesapply/reconciler.go:602]: nit: this function should probably be renamed now that it's not just ImageMaps

File: internal/engine/buildcontrol/image_build_and_deployer.go
Comment (landism) [inline internal/engine/buildcontrol/image_build_and_deployer.go:313]: probably worth linking to the GH issue here

**Review by landism — APPROVED** [review]
LGTM!

**Pull request merged**
